ihatemoney/forms.py

```diff
@@ -142,6 +147,23 @@ def save(self):
         )
         return project
 
+    def validate_default_currency(form, field):
+        project = Project.query.get(form.id.data)
+        if (
+            project is not None
+            and field.data == CurrencyConverter.no_currency
+            and project.has_multiple_currencies()
+        ):
+            raise ValidationError(
+                _(
+                    (
+                        "This project already uses different currencies"
+                        "and can't be set to 'No Currency'"
+                    )
+                )
+            )
+        pass
+
```

Comment:
Where is this used? I don't find any call to this function.

Small suggestion for the error message: "This project cannot be set to 'no currency' because it contains bills in multiple currencies."

Reply:
This is used by WTForms. A function called `validate_<FIELD_NAME>` will be called to validate `FIELD_NAME`.

Thanks for the suggestion, it will put emphasis on the error itself.

Reply:
Ah, right!